ats/atsMachines/fluxScheduled.py

```diff
@@ -0,0 +1,115 @@
+#ATS:slurm56 SELF FluxScheduled 3
+#ATS:toss_4_x86_64_ib SELF FluxScheduled 16
+
+"""
+This module defines the ``FluxScheduled`` class.
+"""
+
+import time
+from math import ceil
+
+import flux
+import flux.job
+
+from ats import terminal
+from ats.tests import AtsTest
+from ats.atsMachines import lcMachines
+
+## Do we need to request an allocation? On fluke if you issue Flux mini run it sends to a compute node automatically
+## but there's no assurances it will be to the _same_ compute nodes for each of ats's "tests" or whatever
+## answer: YES for now ## Should start its own Flux session!! TODO
+
+FLUX_HANDLE = flux.Flux()  ## where do I go?
+
+class FluxScheduled(lcMachines.LCMachineCore):
+    """A class to initialize Flux if necessary and return job statements
+    from ATS tests.
+    """
+
+    def init(self):
+        """Sets ceiling on number of nodes and cores in the allocation.
+        Defines a persistent handle to use to connect to the broker."""
+        self.maxNodes = int(flux.resource.list.resource_list(FLUX_HANDLE).get().up.nnodes)
+        self.maxCores = int(flux.resource.list.resource_list(FLUX_HANDLE).get().up.ncores)
+        self.coresPerNode = int(self.maxCores // self.maxNodes)
+        self.numberTestsRunningMax = self.maxCores
+
+    def examineOptions(self, options):
+        """Optparse (soon argparse) parameters from command-line options
+        for ATS users. Needed for functionality with .ats files.
+        
+        :param options: The options available to a user in test.ats files."""
+        self.exclusive = options.exclusive
+        self.timelimit = options.timelimit
+
+    def calculateCommandList(self, test: AtsTest):
+        """Generates a list of commands to run a test in ATS on a 
+        flux instance.
+        
+        :param test: the test to be run, of type ATSTest. Defined in /ats/tests.py."""
+        ret = "flux mini run".split()
+        np = test.options.get('np', 1)
+        nn = test.options.get('nn', 1)
+        max_time = test.options.get('timelimit', '29m')
+        if np > self.coresPerNode:
+            nn = ceil(np / self.coresPerNode)
+        ret.append(f"-N{nn}")
+        ret.append(f"-n{np}")
+        ret.append(f"-t{max_time}")
```

Comment:
Have you checked that ATS's time limit notation is compatible with Flux's? Otherwise you might need to do some conversion (sounds kind of ugly and not fun)